ProjectSolenodon/Assets/Scripts/LevelManager.cs

```diff
@@ -0,0 +1,61 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using UnityEngine;
+
+/// <summary>
+/// The LevelManager class is used to build scenes that associated to different levels
+/// by instantiating needed game objects.
+/// It also has the ability to serialize and deserialize LevelData objects.
+/// </summary>
+public class LevelManager : MonoBehaviour

```

Comment:
Again, consider renaming, manager is ambiguous.

LevelGenerator
LevelBuilder
PuzzleGenerator
PuzzleSpawner
...